lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -301,6 +320,138 @@ LocaleMatch lookupMatcher(
   return nsStringToU16String([nsStr
       uppercaseStringWithLocale:[[NSLocale alloc] initWithLocaleIdentifier:L]]);
 }
+// Implementation of
+// https://tc39.es/ecma402/#sec-getoption
+// Split into getOptionString and getOptionBool to help readability
+vm::CallResult<llvh::Optional<std::u16string>> getOptionString(
+    const Options &options,
+    const std::u16string &property,
+    llvh::ArrayRef<std::u16string> values,
+    llvh::Optional<std::u16string> fallback,
+    vm::Runtime *runtime) {
+  // 1. Assert type(options) is object
+  // 2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  // 3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    return fallback;
+  }
+  // 4. Assert: type is "boolean" or "string".
+  // 5. If type is "boolean", then
+  // a. Set value to ! ToBoolean(value).
+  // 6. If type is "string", then
+  // a. Set value to ? ToString(value).
+  // 7. If values is not undefined and values does not contain an element equal
+  // to value, throw a RangeError exception.
+  if (!value->second.isString() ||
+      (!values.empty() &&
+       llvh::find(values, value->second.getString()) == values.end())) {
+    return runtime->raiseRangeError(
+        vm::TwineChar16("Value ") +
+        vm::TwineChar16(value->second.getString().c_str()) +
+        vm::TwineChar16(
+            " out of range for Intl.DateTimeFormat options property ") +
+            vm::TwineChar16(property.c_str()));
+  }
+  // 8. Return value.
+  return llvh::Optional<std::u16string>(value->second.getString());
+}
+vm::CallResult<llvh::Optional<bool>> getOptionBool(
+    const Options &options,
+    const std::u16string &property,
+    llvh::Optional<bool> fallback,
+    vm::Runtime *runtime) {
+  //  1. Assert: Type(options) is Object.
+  //  2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  //  3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    return fallback;
+  }
+  if (!value->second.isBool()) {
+    return runtime->raiseTypeError(vm::TwineChar16("Option for property ") + vm::TwineChar16(property.c_str()) + vm::TwineChar16(" is not a bool"));
```

Comment:
These checks and error messages for getOptionString and getOptionBool must be identical except for the type checked.

Reply:
Thanks, I’ve amended it so that both error messages are now the same. I don’t think the llvh::find check needs to be used for getOptionBool, as a values vector isn’t passed in?

Reply:
Yeah, you right about llvh::find, I didn't see that values vector is not there.